teeracle/src/lib.rs

```diff
@@ -151,10 +171,42 @@ pub mod pallet {
 			Ok(())
 		}
 
+		#[pallet::weight(<T as Config>::WeightInfo::update_oracle())]
+		pub fn update_oracle(
+			origin: OriginFor<T>,
+			oracle_name: OracleDataName,
+			data_source: DataSource,
+			new_blob: OracleDataBlob<T>,
+		) -> DispatchResultWithPostInfo {
+			let signer = ensure_signed(origin)?;
+			<pallet_teerex::Pallet<T>>::is_registered_enclave(&signer)?;
+			let signer_index = <pallet_teerex::Pallet<T>>::enclave_index(signer);
+			let signer_enclave = <pallet_teerex::Pallet<T>>::enclave(signer_index)
+				.ok_or(pallet_teerex::Error::<T>::EmptyEnclaveRegistry)?;
+
+			ensure!(
+				Self::is_whitelisted(&data_source, signer_enclave.mr_enclave),
+				<Error<T>>::ReleaseNotWhitelisted
```

Comment:
I wonder about the error's name here, but I guess this was like this before?

Reply:
Yea this is me just using the same convention which was in place before to keep consistency.